storm-core/src/jvm/backtype/storm/blobstore/BlobStoreAclHandler.java

```diff
@@ -0,0 +1,382 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.security.auth.Subject;
+import java.security.Principal;
+import java.util.*;
+
+/**
+ * Provides common handling of acls for Blobstores.
+ * Also contains some static utility functions related to Blobstores.
+ */
+public class BlobStoreAclHandler {
+  public static final Logger LOG = LoggerFactory.getLogger(BlobStoreAclHandler.class);
+  private final IPrincipalToLocal _ptol;
+
+  public static final int READ = 0x01;
+  public static final int WRITE = 0x02;
+  public static final int ADMIN = 0x04;
+  public static final List<AccessControl> WORLD_EVERYTHING =
+      Arrays.asList(new AccessControl(AccessControlType.OTHER, READ | WRITE | ADMIN));
+  public static final List<AccessControl> DEFAULT = new ArrayList<AccessControl>();
+  private Set<String> _supervisors;
+  private Set<String> _admins;
+
+  public BlobStoreAclHandler(Map conf) {
+    _ptol = AuthUtils.GetPrincipalToLocalPlugin(conf);
+    _supervisors = new HashSet<String>();
+    _admins = new HashSet<String>();
+    if (conf.containsKey(Config.NIMBUS_SUPERVISOR_USERS)) {
+      _supervisors.addAll((List<String>)conf.get(Config.NIMBUS_SUPERVISOR_USERS));
+    }
+    if (conf.containsKey(Config.NIMBUS_ADMINS)) {
+      _admins.addAll((List<String>)conf.get(Config.NIMBUS_ADMINS));
+    }
+  }
+
+  private static AccessControlType parseACLType(String type) {
+    if ("other".equalsIgnoreCase(type) || "o".equalsIgnoreCase(type)) {
+      return AccessControlType.OTHER;
+    } else if ("user".equalsIgnoreCase(type) || "u".equalsIgnoreCase(type)) {
+      return AccessControlType.USER;
+    }
+    throw new IllegalArgumentException(type+" is not a valid access control type");
+  }
+
+  private static int parseAccess(String access) {
+    int ret = 0;
+    for (char c: access.toCharArray()) {
+      if ('r' == c) {
+        ret = ret | READ;
+      } else if ('w' == c) {
+        ret = ret | WRITE;
+      } else if ('a' == c) {
+        ret = ret | ADMIN;
+      } else if ('-' == c) {
+        //ignored
+      } else {
+        throw new IllegalArgumentException("");
+      }
+    }
+    return ret;
+  }
+
+  public static AccessControl parseAccessControl(String str) {
+    String[] parts = str.split(":");
+    String type = "other";
+    String name = "";
+    String access = "-";
+    if (parts.length > 3) {
+      throw new IllegalArgumentException("Don't know how to parse "+str+" into an ACL value");
+    } else if (parts.length == 1) {
+      type = "other";
+      name = "";
+      access = parts[0];
+    } else if (parts.length == 2) {
+      type = "user";
+      name = parts[0];
+      access = parts[1];
+    } else if (parts.length == 3) {
+      type = parts[0];
+      name = parts[1];
+      access = parts[2];
+    }
+    AccessControl ret = new AccessControl();
+    ret.set_type(parseACLType(type));
+    ret.set_name(name);
+    ret.set_access(parseAccess(access));
+    return ret;
+  }
```

Comment:
This allows some shortcuts in giving ACL entries.
`rwa` or `::rwa` -> `other::rwa`
`alice:r` or `:alice:r` -> `user:alice:r`
`u:bob:rwa` -> `user:bob:rwa`

Looks good.  Maybe we could add it to the CLI help?  If it is already clear enough, then we do not need to.